### PR TITLE
Add metadata field to retire_credits function

### PR DESCRIPTION
**Pull request by haveanicedavid**
references: regen-network/regen-registry#646
[Frontend PR](https://github.com/regen-network/regen-web/pull/747)

### CODE REVIEW

File: sql/R__retire_credits.sql [hunk 1]
```diff
@@ -2,7 +2,8 @@ create or replace function retire_credits(
   vintage_id uuid,
   buyer_wallet_id uuid,
   address_id uuid,
-  units numeric
+  units numeric,
+  metadata jsonb
```
Comment (blushi) [inline sql/R__retire_credits.sql:6]:
```suggestion
  metadata jsonb default '{}'
```

File: sql/R__transfer_credits.sql [hunk 2]
```diff
@@ -199,7 +199,7 @@ begin
 
   -- retire credits if auto_retire true
   if auto_retire = true then
-    perform retire_credits(vintage_id, buyer_wallet_id, address_id, units);
+    perform retire_credits(vintage_id, buyer_wallet_id, address_id, units, default);
```
Comment (blushi) [inline sql/R__transfer_credits.sql:202]: this makes the tests fail, if you apply the change from above, you should be able to just use:
```suggestion
    perform retire_credits(vintage_id, buyer_wallet_id, address_id, units);
```

**Review by blushi — CHANGES_REQUESTED** [review]

File: sql/R__retire_credits.sql (hunk 1 above)
[the remark this answers is not in the document]
Comment (haveanicedavid) [reply sql/R__retire_credits.sql]: This fixed an issue I kept bumping into and makes sense after reading through docs - thank you!

**Comment by haveanicedavid** [comment]
@blushi just updated - tests now pass

**Review by blushi — APPROVED** [review]
Approving now that I've changed the base branch to be `dev`, please make sure to submit registry-server PRs against `dev` first.

**Review by mhagel — APPROVED** [review]

**Pull request merged**